MetaMask/metamask-extension
# Adding Icon component and removing BaseIcon component

## Explanation

Adding new `Icon` component that uses the css property `mask`. It also includes:
- icon svgs `ui/components/component-library/icons/svgs`
- scripts to generate the icon names and build process updates to store them in an environment variable, `process.env.ICON_NAMES`

## More Information
This is a revised version of the [react svg component version](https://github.com/MetaMask/metamask-extension/pull/15555) after [some security and build step concerns were raised](https://consensys.slack.com/archives/C0354T27M5M/p1661359820969389)

* Fixes #15098

## Screenshots/Screencaps


https://user-images.githubusercontent.com/8112138/189249160-0e86d3a3-8c14-4cbc-8736-4ea1232b6bd3.mov

### Browser testing

[`mask-image` has reasonable browser compatibility with vendor prefixing](https://caniuse.com/?search=mask-image)

<img width="1372" alt="Screen Shot 2022-09-15 at 1 55 44 PM" src="https://user-images.githubusercontent.com/8112138/190507162-dadabf77-8c1b-41b1-b799-f4f49a7fa0da.png">


**Chrome**

<img width="871" alt="Screen Shot 2022-09-14 at 2 16 22 PM" src="https://user-images.githubusercontent.com/8112138/190504873-8d95013c-d17c-40dd-95a0-f1c8473ef6a4.png">
<img width="768" alt="Screen Shot 2022-09-14 at 2 16 44 PM" src="https://user-images.githubusercontent.com/8112138/190504877-62a726c5-f1a6-4332-be69-dad3de1e3dcf.png">

**Firefox**
<img width="618" alt="Screen Shot 2022-09-15 at 1 49 52 PM" src="https://user-images.githubusercontent.com/8112138/190507243-d32c6752-ca2c-420b-bb57-adf6a46a12dc.png">


## Manual Testing Steps

Storybook
- Go to the latest build of storybook in this PR
- Search for `Icon`
- Check the stories, controls and documentation

Build
- Check out this branch
- In the terminal run `yarn start`
- Should see ICON_NAMES env car successfully generated
![Screen Shot 2022-09-15 at 12 52 42 PM](https://user-images.githubusercontent.com/8112138/190496436-45a078e0-757d-42f4-baae-d715c680fdd4.png)


## Pre-Merge Checklist


### + If there are functional changes:
